frontend/pages/hosts/components/ScriptDetailsModal/ScriptDetailsModal.tsx

```diff
@@ -93,20 +96,30 @@ const ScriptDetailsModal = ({
   const { currentUser } = useContext(AppContext);
   const { renderFlash } = useContext(NotificationContext);
 
+  // handle multiple possibilities for `selectedScriptDetails`
+  let scriptId: number | null = null;
+  if (selectedScriptDetails) {
+    if ("script_id" in selectedScriptDetails) {
+      scriptId = selectedScriptDetails.script_id;
+    } else if ("id" in selectedScriptDetails) {
+      scriptId = selectedScriptDetails.id;
+    }
+  }
+
```

Comment:
I think it would be better to export the expected type from here (as something like `ScriptDetailsScript`) and have `RunScriptBatchPaginatedList` set `selectedScriptDetails` to an instance of that, rather than having to support different types in here, what do you think?